friends/src/App.js

```diff
@@ -0,0 +1,70 @@
+
+import React, { Component } from 'react';
+import './App.css';
+import { connect } from 'react-redux';
+import { getFriends, addFriend, deleteFriend, updateFriend } from './actions'
+import Friends from './components/Friends';
+import FriendForm from './components/CreateFriendForm';
+import FriendUpdateForm from './components/UpdateFriendForm';
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isUpdate: false,
+      upDatingFriend: {}
+    }
+  }
+
+  componentDidMount() {
+    this.props.getFriends()
+  }
+
+  handleUpdate = event => {
+    this.setState({
+      isUpdate: true,
+      upDatingFriend: this.props.friends.filter(friend => friend.id === event.target.id)[0]
```

Comment:
This filter method is not working correctly because `event.target.id` is evaluating as a _string_ while `friend.id` is evaluating as _number_. You could cast one to a string or number for the condition to work semantically. For instance:
```
handleUpdate = event => {
  const id = Number(event.target.id);
  this.setState({
    isUpdate: true,
    upDatingFriend: this.props.friends.filter(friend => friend.id === id)[0]
    })
  }
```